store/views.py

```diff
@@ -57,26 +59,86 @@ def loanBookView(request):
     response_data = {
         'message': None,
     }
-    '''
-    Check if an instance of the asked book is available.
-    If yes, then set the message to 'success', otherwise 'failure'
-    '''
-    # START YOUR CODE HERE
-    book_id = None # get the book id from post data
+    
+    data = request.POST
+    if request.method=='POST':
+        bid = data.get('bid','')
+    book_id = bid 
 
+    bookcopy = BookCopy.objects.filter(book=book_id,status=True)
+
+    if len(bookcopy)==0:
+        response_data['message'] = 'failure'
+    else:    
+        bookcopy[0].borrower = request.user
+        bookcopy[0].borrow_date = date.today()
+        bookcopy[0].status = False
+        bookcopy[0].save()
+        response_data['message'] = 'success'
 
     return JsonResponse(response_data)
 
-'''
-FILL IN THE BELOW VIEW BY YOURSELF.
-This view will return the issued book.
-You need to accept the book id as argument from a post request.
-You additionally need to complete the returnBook function in the loaned_books.html file
-to make this feature complete
-''' 
+
 @csrf_exempt
 @login_required
 def returnBookView(request):
-    pass
+    response_data = {
+        'message': None,
+    }
+
+    data = request.POST
+    if request.method=='POST':
+        bid = data.get('bid','')
+    book_id = bid 
+
+    print ("CONSOLE LOG")
+    bookcopy = BookCopy.objects.filter(pk=book_id)
+
+    if len(bookcopy)==0:
+        response_data['message'] = 'failure'
+    else:    
+        bookcopy[0].borrower = None
+        bookcopy[0].borrow_date = None
+        bookcopy[0].status = True
+        bookcopy[0].save()
+        response_data['message'] = 'success'
+
+    return JsonResponse(response_data)
+
+@csrf_exempt
+@login_required
+def rateBookView(request):
+    
+    if request.method == "POST":
+        data = request.POST
+        bid=data.get('bid','')
+        rate=data.get('rate',0.0)
+        print(bid)
+        print(rate)
+        book = Book.objects.get(pk=bid)
+        oldRating=UserRating.objects.filter(user=request.user,book=book)
+        rating=UserRating()
+        rating.book=book
+        rating.user=request.user
+        rating.rating=rate
+        oldRating.delete()
+        rating.save()
```

Comment:
You've not put a backend validation on the rating, so the user can easily put invalid values of rating.

You could have updated the rating rather than deleting and then saving it.

Reply:
But user only has 1-10 option of rating in a dropdown menu? Am I getting this wrong?